### PR TITLE
Miscellaneous commits from RmlUi branch

### PR DESCRIPTION
**Pull request by slipher**
Ones which don't directly relate to RmlUi migration.

### CODE REVIEW

File: src/cgame/cg_snapshot.cpp [hunk 1]
```diff
@@ -124,7 +124,7 @@ void CG_SetInitialSnapshot( snapshot_t *snap )
 	}
 
 	// Need this check because the initial weapon for spec isn't always WP_NONE
-	if ( CG_MyTeam() == TEAM_NONE )
```
Comment (bmorel) [inline src/cgame/cg_snapshot.cpp:127]: I checked what CG_MyTeam() does, and it checks `cg.predictedPlayerState.persistant[PERS_TEAM]`.
Why the change? (I don't say this is wrong, I focused my efforts on sgame so far)

BTW we should change `G_Team( gentity_t* )` so that it becomes `BG_Team( playerState_t const& )` to avoid that kind of things... I thought this already existed, but after checking, nope, it does not.

**Review by bmorel — APPROVED** [review]
LGTM

File: src/cgame/cg_snapshot.cpp (hunk 1 above)
Comment (slipher) [reply src/cgame/cg_snapshot.cpp:127]: It doesn't work with predictedPlayerState - it's all zero the first time this executes

Comment (bmorel) [reply src/cgame/cg_snapshot.cpp:127]: I see. Thanks for explanations.

**Comment by bmorel** [comment]
Concerning the revert, it appears that it's the reason why packages does not work at all, even with custom maprotation file.
I do know know the reason, but here, neither the engine nor the sgame module gives any hint about what happened.

I think this commit should be reverted quickly, possibly individually from this PR, since it breaks master.

**Comment by slipher** [comment]
@necessarily-equal  @ing in case you want to investigate the reverted one
